src/Microsoft.Azure.SignalR.Management/DependencyInjectionExtensions.cs

```diff
@@ -108,13 +108,25 @@ private static IServiceCollection AddSignalRServiceCore(this IServiceCollection
         /// <summary>
         /// Adds product info to <see cref="ServiceManagerOptions"/>
         /// </summary>
-        [EditorBrowsable(EditorBrowsableState.Never)]
         public static IServiceCollection WithAssembly(this IServiceCollection services, Assembly assembly)
         {
             var productInfo = ProductInfo.GetProductInfo(assembly);
             return services.Configure<ServiceManagerOptions>(o => o.ProductInfo = productInfo);
         }
 
+        /// <summary>
+        /// Allows functions extensions to add additional product info.
+        /// </summary>
+        public static IServiceCollection AddUserAgent(this IServiceCollection services, string userAgent)
+        {
+            if (userAgent is null)
+            {
+                throw new ArgumentNullException(nameof(userAgent));
+            }
+
+            return services.PostConfigure<ServiceManagerOptions>(o => o.ProductInfo += userAgent);
```

Comment:
what if user call WithUserAgent(..).WithAssembly()?

Reply:
The post configure actions are always executed after configure actions.

Reply:
throw if ProductInfo is null?

Reply:
The `AddSignalRServiceCore` makes sure `ProductInfo` is not null. Will update to "string. Empty + userAgent" if product is null.

Reply:
I think throwing if null is better now as it can serve as a validation as well.